src/shared/components/VideoPreviewBase/VideoPreviewBase.tsx

```diff
@@ -0,0 +1,217 @@
+  CoverPlayIcon,
+  MetaText,
+  ProgressBar,
+  ProgressOverlay,
+  StyledAvatar,
+  TitleHeader,
+  Anchor,
+  CoverPlaceholder,
+  SpacedPlaceholder,
+} from './VideoPreviewBase.styles'
+import { formatVideoViewsAndDate } from '@/utils/video'
+import { formatDurationShort } from '@/utils/time'
+import useResizeObserver from 'use-resize-observer'
+import { transitions } from '@/shared/theme'
+import { SwitchTransition, CSSTransition } from 'react-transition-group'
+import { Placeholder } from '..'
+
+export type VideoPreviewBaseMetaProps = {
+  showChannel?: boolean
+  showMeta?: boolean
+  main?: boolean
+  onClick?: (e: React.MouseEvent<HTMLElement>) => void
+  onChannelClick?: (e: React.MouseEvent<HTMLElement>) => void
+  onCoverResize?: (width: number | undefined, height: number | undefined) => void
+}
+export type VideoPreviewBaseProps = {
+  title?: string
+  channelHandle?: string
+  channelAvatarUrl?: string | null
+  createdAt?: Date
+  duration?: number
+  // video watch progress in percent (0-100)
+  progress?: number
+  views?: number | null
+  thumbnailUrl?: string
+  isLoading?: boolean
+  videoHref?: string
+  channelHref?: string
+  className?: string
+} & VideoPreviewBaseMetaProps
+
+export const MIN_VIDEO_PREVIEW_WIDTH = 300
+const MAX_VIDEO_PREVIEW_WIDTH = 600
+const MIN_SCALING_FACTOR = 1
+const MAX_SCALING_FACTOR = 1.375
+// Linear Interpolation, see https://en.wikipedia.org/wiki/Linear_interpolation
+const calculateScalingFactor = (videoPreviewWidth: number) =>
+  MIN_SCALING_FACTOR +
+  ((videoPreviewWidth - MIN_VIDEO_PREVIEW_WIDTH) * (MAX_SCALING_FACTOR - MIN_SCALING_FACTOR)) /
+    (MAX_VIDEO_PREVIEW_WIDTH - MIN_VIDEO_PREVIEW_WIDTH)
+
+const VideoPreviewBase: React.FC<VideoPreviewBaseProps> = ({
+  title,
+  channelHandle,
+  channelAvatarUrl,
+  createdAt,
+  duration,
+  progress = 0,
+  views,
+  thumbnailUrl,
+  onCoverResize,
+  channelHref,
+  videoHref,
+  isLoading = true,
+  showChannel = true,
+  showMeta = true,
+  main = false,
+  onChannelClick,
+  onClick,
+  className,
+}) => {
+  const [scalingFactor, setScalingFactor] = useState(MIN_SCALING_FACTOR)
+  const { ref: imgRef } = useResizeObserver<HTMLImageElement>({
+    onResize: (size) => {
+      const { width: videoPreviewWidth, height: videoPreviewHeight } = size
+      if (onCoverResize) {
+        onCoverResize(videoPreviewWidth, videoPreviewHeight)
+      }
+      if (videoPreviewWidth && !main) {
+        setScalingFactor(calculateScalingFactor(videoPreviewWidth))
+      }
+    },
+  })
+
+  const displayChannel = showChannel && !main
+  const clickable = (!!onClick || !!videoHref) && !isLoading
+  const channelClickable = (!!onChannelClick || !!channelHref) && !isLoading
+  const handleChannelClick = (e: React.MouseEvent<HTMLElement>) => {
+    if (!onChannelClick) {
+      return
+    }
+    e.stopPropagation()
+    onChannelClick(e)
+  }
+  const handleAnchorClick = (href?: string) => (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
```

Comment:
Like in the carousels, let's indicate a difference between those handlers (this is a higher order function) and regular handlers (which are just functions)